specs/slider/Spec.md

```diff
@@ -32,167 +43,177 @@ Sliders are great for adjusting settings that reflect intensity levels (volume,
+E.g. if min=0, max=10, step=2, value=3.5, then UP arrow key will give us 4 and Down arrow key will give us 2
+Value can not exceed the min/max limits. If value is > or < than min/max it is automatically set to corresponding min/max.
 
-Slider should have a focus state. Focus state looks like a square border around handle. We place focus around handle because in range slider there is no other options and we need to keep consistency in all components. 
-![sliderFocus](./assets/sliderFocus.png)
+If slider has a 'step' prop, handle should move across the slider bar only according to the step.  
 
-reference links: 
-https://www.paciellogroup.com/blog/2008/05/aria-slider-part-1/
-https://www.paciellogroup.com/blog/2008/06/aria-slider-part-2/
+#### Validation
 
+| Case                          | Handling          |
+| ----------------------------- | ----------------- |
+| `value` is less then `min`    | set to `min`      |
+| `value` is greater than `max` | set to `max`      |
+| `value` is empty              | set to `min` OR 0 |
 
 
-## Examples
 
-#### Simple
+#### Edge case handling
 
-The `defaultValue` property sets the initial position of the slider. 
+| Case                                     | Handling                                 | Example                                  |
+| ---------------------------------------- | ---------------------------------------- | ---------------------------------------- |
+| value is out of `step`                   | keep current value & display handle at the position that represents this value | `min=0` `max=10` `step=2` `value=3.5`. Handle is displayed at the position that represents 3.5. When user changes value (keyboard / mouse) we move it to the next expected, which is 4 or 2 |
+| `step` is not set & `displayMarks` is set to true | show marks at the beginning and at the end of the slider |                                          |
 
-![simpleExample](./assets/simpleExample.png)
 
 
+### Input Methods 
 
-#### Stepped
+#### Keyboard
 
-By default, the slider is continuous. The step property causes the slider to move in discrete increments.
+| Keys                                 | Action                      |
+| ------------------------------------ | --------------------------- |
+| up / right arrow key                 | increase value              |
+| left / down arrow key                | decrease value              |
+| home ( fn/ctrl + left arrow key)     | set min value               |
+| end ( fn/ctrl + left arrow key)      | set min value               |
+| page up (fn/ctrl + up arrow key)     | increase value by X         |
+| page down (fn/ctrl + down arrow key) | decrease value by X         |
+| tab                                  | moves to next element       |
+| shift + tab                          | moves to previous element   |
+| enter                                | -                           |
+| esc                                  | removes focus (if in focus) |
```

Comment:
Escape does not remove focus from a slider. You can see this when looking at a native input of type range. See also [here](https://www.w3.org/TR/wai-aria-practices/#kbd_shortcuts_assignments_opsys_conflicts), when using system keys, do so only if the need is defined by the w3 draft, otherwise behavior would be indeterminate.